File: app/src/main/java/com/example/ui/components/FinancialOverviewCard.kt
package com.example.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccountBalance
import androidx.compose.material.icons.filled.ArrowDownward
import androidx.compose.material.icons.filled.ArrowUpward
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.core.theme.DesignTokens
import com.example.core.util.CurrencyFormatter
import com.example.domain.model.NetWorthSummary
import com.example.ui.theme.GreenIncome
import com.example.ui.theme.RedExpense
import com.example.ui.theme.TealAccent

@Composable
fun FinancialOverviewCard(
    summary: NetWorthSummary,
    isArabic: Boolean = false
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp, vertical = 8.dp),
        shape = RoundedCornerShape(DesignTokens.RadiusLarge),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
    ) {
        Column(
            modifier = Modifier.padding(20.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column {
                    Text(
                        text = if (isArabic) "إجمالي صافي الثروة" else "TOTAL NET WORTH",
                        style = MaterialTheme.typography.labelSmall.copy(
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            letterSpacing = 1.sp
                        )
                    )
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = CurrencyFormatter.format(summary.netWorth, summary.currency),
                        style = MaterialTheme.typography.headlineLarge.copy(
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurface,
                            letterSpacing = (-0.5).sp
                        )
                    )
                }

                Box(
                    modifier = Modifier
                        .size(44.dp)
                        .clip(RoundedCornerShape(12.dp))
                        .background(TealAccent.copy(alpha = 0.15f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.TrendingUp,
                        contentDescription = if (isArabic) "صافي الثروة" else "Net Worth",
                        tint = TealAccent,
                        modifier = Modifier.size(24.dp)
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))
            HorizontalDivider(color = MaterialTheme.colorScheme.outline.copy(alpha = 0.2f))
            Spacer(modifier = Modifier.height(16.dp))

            // Breakdown Grid
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                OverviewColumnItem(
                    label = if (isArabic) "النقد والحسابات" else "Cash & Accounts",
                    amount = summary.totalCashAndAccounts,
                    currency = summary.currency,
                    color = MaterialTheme.colorScheme.onSurface
                )
                OverviewColumnItem(
                    label = if (isArabic) "إجمالي الأصول" else "Total Assets",
                    amount = summary.totalAssetsValue,
                    currency = summary.currency,
                    color = GreenIncome
                )
                OverviewColumnItem(
                    label = if (isArabic) "الديون والالتزامات" else "Debts / Payables",
                    amount = summary.totalLiabilitiesAndPayables,
                    currency = summary.currency,
                    color = RedExpense
                )
            }
        }
    }
}

@Composable
fun OverviewColumnItem(
    label: String,
    amount: Double,
    currency: String,
    color: Color
) {
    Column {
        Text(
            text = label,
            style = MaterialTheme.typography.labelSmall.copy(
                fontSize = 11.sp,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        )
        Spacer(modifier = Modifier.height(2.dp))
        Text(
            text = CurrencyFormatter.formatCompact(amount, currency),
            style = MaterialTheme.typography.titleMedium.copy(
                fontWeight = FontWeight.Bold,
                color = color
            )
        )
    }
}

File: lib/ui/components/financial_overview_card.dart
import 'package:flutter/material.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/design_tokens.dart';
import '../../core/util/currency_formatter.dart';
import '../../domain/models/domain_models.dart';

class FinancialOverviewCard extends StatelessWidget {
  final NetWorthSummary netWorth;
  final bool isArabic;

  const FinancialOverviewCard({
    super.key,
    required this.netWorth,
    required this.isArabic,
  });

  @override
  Widget build(BuildContext context) {
    return Card(
      margin: const EdgeInsets.all(DesignTokens.spaceMedium),
      color: AppColors.slateDarkCard,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(DesignTokens.radiusLarge),
        side: const BorderSide(color: AppColors.slateBorder, width: DesignTokens.borderWidth),
      ),
      child: Padding(
        padding: const EdgeInsets.all(DesignTokens.spaceLarge),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text(
                  isArabic ? "صافي الثروة" : "Net Worth",
                  style: const TextStyle(fontSize: 14, color: AppColors.textSecondaryDark),
                ),
                const Icon(Icons.shield_outlined, color: AppColors.tealAccent, size: 20),
              ],
            ),
            const SizedBox(height: DesignTokens.spaceSmall),
            Text(
              CurrencyFormatter.format(netWorth.netWorth, "SAR"),
              style: const TextStyle(fontSize: 28, fontWeight: FontWeight.bold, color: AppColors.goldAccent),
            ),
            const SizedBox(height: DesignTokens.spaceLarge),
            const Divider(color: AppColors.slateBorder),
            const SizedBox(height: DesignTokens.spaceMedium),
            Row(
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(isArabic ? "النقد والحسابات" : "Cash & Accounts", style: const TextStyle(fontSize: 11, color: AppColors.textMutedDark)),
                      Text(CurrencyFormatter.formatCompact(netWorth.totalCashAndAccounts, "SAR"), style: const TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimaryDark)),
                    ],
                  ),
                ),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(isArabic ? "قيمة الأصول" : "Assets Value", style: const TextStyle(fontSize: 11, color: AppColors.textMutedDark)),
                      Text(CurrencyFormatter.formatCompact(netWorth.totalAssetsValue, "SAR"), style: const TextStyle(fontWeight: FontWeight.bold, color: AppColors.purpleAsset)),
                    ],
                  ),
                ),
              ],
            ),
            const SizedBox(height: DesignTokens.spaceMedium),
            Row(
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(isArabic ? "ديون لك (مستحقات)" : "Receivables", style: const TextStyle(fontSize: 11, color: AppColors.textMutedDark)),
                      Text(CurrencyFormatter.formatCompact(netWorth.totalReceivables, "SAR"), style: const TextStyle(fontWeight: FontWeight.bold, color: AppColors.greenIncome)),
                    ],
                  ),
                ),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(isArabic ? "ديون عليك (التزامات)" : "Liabilities", style: const TextStyle(fontSize: 11, color: AppColors.textMutedDark)),
                      Text(CurrencyFormatter.formatCompact(netWorth.totalLiabilitiesAndPayables, "SAR"), style: const TextStyle(fontWeight: FontWeight.bold, color: AppColors.redExpense)),
                    ],
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
